tests/lightning_qubit/test_measurements_class.py

```diff
@@ -515,45 +515,60 @@ def test_single_return_value(self, measurement, observable, lightning_sv, tol):
             qml.SparseHamiltonian(get_sparse_hermitian_matrix(2**4), wires=range(4)),
         ),
     )
-    def test_double_return_value(self, measurement, obs0_, obs1_, lightning_sv, tol):
-        if measurement is qml.probs and isinstance(
-            obs0_,
-            (qml.ops.Sum, qml.ops.SProd, qml.ops.Prod, qml.Hamiltonian, qml.SparseHamiltonian),
+    def test_double_return_value(self, shots, measurement, obs0_, obs1_, lightning_sv, tol):
+        skip_list = (
+            qml.ops.Sum,
+            qml.ops.SProd,
+            qml.ops.Prod,
+            qml.Hamiltonian,
+            qml.SparseHamiltonian,
+        )
+        if measurement is qml.probs and (
+            isinstance(obs0_, skip_list) or isinstance(obs1_, skip_list)
         ):
             pytest.skip(
                 f"Observable of type {type(obs0_).__name__} is not supported for rotating probabilities."
             )
 
-        if measurement is qml.probs and isinstance(
-            obs1_,
-            (qml.ops.Sum, qml.ops.SProd, qml.ops.Prod, qml.Hamiltonian, qml.SparseHamiltonian),
-        ):
-            pytest.skip(
-                f"Observable of type {type(obs1_).__name__} is not supported for rotating probabilities."
-            )
-
         n_qubits = 4
         n_layers = 1
         np.random.seed(0)
         weights = np.random.rand(n_layers, n_qubits, 3)
         ops = [qml.Hadamard(i) for i in range(n_qubits)]
         ops += [qml.StronglyEntanglingLayers(weights, wires=range(n_qubits))]
         measurements = [measurement(op=obs0_), measurement(op=obs1_)]
-        tape = qml.tape.QuantumScript(ops, measurements)
+        tape = qml.tape.QuantumScript(ops, measurements, shots=shots)
 
-        expected = self.calculate_reference(tape, lightning_sv)
-        if len(expected) == 1:
-            expected = expected[0]
         statevector = lightning_sv(n_qubits)
         statevector = statevector.get_final_state(tape)
         m = LightningMeasurements(statevector)
-        result = m.measure_final_state(tape)
+
+        skip_list = (
+            qml.ops.Sum,
+            qml.Hamiltonian,
+            qml.SparseHamiltonian,
+        )
+        if (
+            (measurement is qml.expval or measurement is qml.var)
+            and shots is not None
+            and (isinstance(obs0_, skip_list) or isinstance(obs1_, skip_list))
+        ):
+            with pytest.raises(TypeError):
+                _ = m.measure_final_state(tape)
+            return
+        else:
+            result = m.measure_final_state(tape)
+
+        expected = self.calculate_reference(tape, lightning_sv)
+        if len(expected) == 1:
+            expected = expected[0]
 
         assert isinstance(result, Sequence)
         assert len(result) == len(expected)
         # a few tests may fail in single precision, and hence we increase the tolerance
+        dtol = tol if shots is None else max(tol, 1.0e-2)
```

Comment:
It is difficult to write a stochastic circuit tests parametrized over observables that will pass with low tolerance because a measurement in the z-basis that comes at ~1, will come out ~0 in the y-basis, say. In my experience, for a measurement ~1 one needs about 10**(2n) shots to safely get an n-digit accuracy, and this is even worse if you're trying to match ~0. I think this is why PennyLane tests use `TOL_STOCHASTIC = 0.05`, and here I tried to push it down to 0.01 by increasing the shots to 1e6 and adding 5 flaky runs. We could write taylored tests with a lower tolerance, but I think I'm catching more bugs by having a higher tolerance and combinatorially testing everything (for example caught one again yesterday flaging only obs1_obs4 combinations).